api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -171,4 +171,27 @@ default Iterable<DeleteFile> removedDeleteFiles(FileIO io) {
   default Integer schemaId() {
     return null;
   }
+
+  /**
+   * The row-id of the first newly added row in this snapshot. The all rows added in this snapshot
```

Comment:
maybe add something to highlight that this is valid only for newly added rows. Updated rows can have smaller row-id.

Maybe (disclaimer English is not my first language):
```
The all newly added rows inserted in this snapshot will have a row-id value of greater than or equal to this value. Only updated rows can have smaller row-id in this snapshot.
```